README.md

```diff
@@ -36,21 +36,47 @@ $ make
 ```
 This builds the plugin in `/build/rootfs`
 
+Unit testing:
+```
+$ make test-small
+```
+
 ### Configuration and Usage
 * Set up the [snap framework](https://github.com/intelsdi-x/snap/blob/master/README.md#getting-started)
 * Ensure `$SNAP_PATH` is exported
 `export SNAP_PATH=$GOPATH/src/github.com/intelsdi-x/snap/build`
 
 ## Documentation
-This Snap processor plugin calculates statistics over a sliding window. Currently, the plugin calculates the mean, median, standard deviation, variance, 95th-percentile and 99th-percenticle over a sliding window. 
-
-Note: This Snap processor plugin changes the metric data type to `map[string]float64`. Any Snap publisher plugin used with this plugin should register the data type with gob. 
-
-```
-import "encoding/gob"
-
-gob.Register(map[string]float64{})
-```
+This Snap processor plugin calculates statistics over a sliding window. Currently, the plugin calculates the following statistics over a sliding window:
+1) Count - length of the buffer
+2) Sum - total sum of data values in the buffer
+3) Mean - average of data values 
+4) Median - middle data value in a sorted buffer 
+5) Minimum - minimum of all data values  
+6) Maximum - maximum of all data values 
+7) Range - difference of largerst and smallest values 
+8) Standard deviation - measure that quantifies the amount of varitation of a dataset 
+9) Variance - measures how far a set of numbers are spread out from their mean
+10) Mode - the number which appears most often in a dataset
+11) Kurtosis - measures the degree of peakedness of a distribution
+12) Skewness - measure of a dataset's symmetry or lack of symmetry
+13) Trimean - measure of a probability distribution's location as a weighted average of the distribution's median and its two quartiles 
+14) First quartile - middle number between the smallest number and the median of the dataset
+15) Third quartile - median of the upper half of the dataset
+16) Quartile range - measure of variability based on dividing a dataset into quartiles
+17) 2nd-percentile - value below which 2% observations may be found 
+18) 9th-percentile - value below which 9% observations may be found 
+19) 25th-percentile - value below which 25% observations may be found 
+20) 75th-percentile - value below which 75% observations may be found 
+21) 91st-percentile - value below which 91% observations may be found  
+22) 95th-percentile - value below which 95% observations may be found 
+23) 98th-percentile - value below which 98% observations may be found 
+24) 99th-percenticle - value below which 99% observations may be found 
```

Comment:
All these percentile definitions are hard to understand. Rewriting them. Aren't they observations at percentile?